audiolat: Made it possible to have the begin signal split over severale buffers

JohanBlome opened the pull request.

JohanBlome committed:
Made it possible to have the begin signal split over severalw buffers
if the signal is longer than the buffer received by the callback.

Removed some unnecessary logs.

Signed-off-by: Johan Blome <jblome@fb.com>

JohanBlome committed:
audiolat: Fixed signal spanning callbacks

* Allow both begin and end signal to span several callbacks
* corrected logs
* minimize allocation of silent data

Signed-off-by: Johan Blome <jblome@fb.com>

JohanBlome committed:
audiolat: Javaaudio and extra args

Added javaaudio playback and record.
Added a few arguments:
- usage, is the number from AudioAttributes.USAGE_XXX
- signal, which can be either "noise" or "chirp", default is chirp
- api, which can be either "aaudio" or "javaaudio"

Signed-off-by: Johan Blome <jblome@fb.com>

chemag commented on app/src/main/jni/audiolat.cpp:
```diff
@@ -213,6 +215,7 @@ Java_com_facebook_audiolat_MainActivity_runAAudio(JNIEnv *env,
   }
   // more of this later...device_id);
   AAudioStreamBuilder_setDeviceId(playout_builder, AAUDIO_UNSPECIFIED);
+  AAudioStreamBuilder_setUsage(playout_builder, usage);
```
why setting the usage in the playout but not in the record stream?

chemag commented on app/src/main/jni/audiolat.cpp:
```diff
@@ -176,6 +176,8 @@ Java_com_facebook_audiolat_MainActivity_runAAudio(JNIEnv *env,
   jint playout_buffer_size = env->GetIntField(settings, fid);
   fid = env->GetFieldID(cSettings, "recordBufferSize", "I");
   jint record_buffer_size = env->GetIntField(settings, fid);
+  fid = env->GetFieldID(cSettings, "usage", "I");
```
why aren't you reading the timeBetweenSignals variable here too?

JohanBlome committed:
Merge branch 'master' of https://github.com/chemag/audiolat into fork

JohanBlome committed:
audiolat: Added support for midi initiated playout

use "-e midi 1" to start midi test
Click on a available device in the list.
Test will automtically close after the stipulated timeout.

Signed-off-by: Johan Blome <jblome@fb.com>

JohanBlome commented:
I have new updates instead.
